### ISSUE TITLE
False Positive:  S1144 Remove the unused private constructor with C#9

### ISSUE THREAD

**Issue by mjeanrichard** [SonarSource/sonar-dotnet#3943]
### Description

The following code triggers rule S1144 (Remove the unused private constructor 'ImmutableThing'.):

```csharp
public class ImmutableThing
{
    private ImmutableThing(string name)
    {
        Name = name;
    }

    public string Name { get; }

    public static ImmutableThing FromNumber(int number)
    {
        return new($"Thing {number}");
    }
}
```

If I specify the Type explicitly after the new keyword the rule is not triggered:
```csharp
public class ImmutableThing
{
    private ImmutableThing(string name)
    {
        Name = name;
    }

    public string Name { get; }

    public static ImmutableThing FromNumber(int number)
    {
        return new ImmutableThing($"Thing {number}");
    }
}
```
### Repro steps

see above.

### Expected behavior

No rule violations in both cases.

### Actual behavior

The first example triggers the rule.

### Known workarounds

Specify the type name (i.e. do not use the Target-typed new expression of C#9)

**Comment by pavel-mikula-sonarsource** [comment]
Hi @mjeanrichard ,

Thank you for reporting this case. I can confirm it as a False Positive. We've recently added a reproducer for this issue in #3921 and we'll add support for C# 9 target typed constructors in [MMF-2225](https://jira.sonarsource.com/browse/MMF-2225) so I'll close this issue for now.
